test/avocado/machineslib.py

```diff
@@ -0,0 +1,166 @@
+import os
+import time
+from subprocess import CalledProcessError
+from testlib_avocado.seleniumlib import SeleniumTest, clickable
+
+SPICE_XML = """
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='spice' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+      <image compression='off'/>
+    </graphics>
+"""
+
+VNC_XML = """
+    <video>
+      <model type='vga' heads='1' primary='yes'/>
+      <alias name='video0'/>
+    </video>
+    <graphics type='vnc' port='5900' autoport='yes' listen='127.0.0.1'>
+      <listen type='address' address='127.0.0.1'/>
+    </graphics>
+"""
+
+CONSOLE_XML = """
+    <console type='file'>
+      <target type='serial' port='0'/>
+      <source path='{log}'/>
+    </console>
+"""
+
+PTYCONSOLE_XML = """
+    <serial type='pty'>
+      <source path='/dev/pts/3'/>
+      <target port='0'/>
+      <alias name='serial0'/>
+    </serial>
+    <console type='pty' tty='/dev/pts/3'>
+      <source path='/dev/pts/3'/>
+      <target type='serial' port='0'/>
+      <alias name='serial0'/>
+    </console>
+"""
+
+DOMAIN_XML = """
+<domain type='qemu'>
+  <name>{name}</name>
+  <vcpu>1</vcpu>
+  <os>
+    <type arch='x86_64'>hvm</type>
```

Comment:
it means that these tests will run just on intel arch doesn't it? what is completely fine inside upstream CI. But do we expect to run these tests also on another arches what supports virtualization?

Reply:
Thanks for pointing this out, I haven't considered that. Seems delete "arch='x86_64'" could work, which will make the guest's arch the same as the host.